### PR TITLE
Generate a single zz_generated.terraformed.go file for the whole group

### LINKED ISSUE #43: Generate a single zz_generated.terraformed.go file for the whole group
We used to put every CRD into its own package, hence each had its own `zz_terraformed` file but now we don't do that. We can generate a single zz_generated.terraformed.go file for the whole group, similar to `zz_managed` and `zz_deepcopy`.

### PR DESCRIPTION
**Pull request by sergenyalcin**
### Description of your changes

Fixes #43

This PR generates a single `terraformed.go` file for the whole API group. For example in provider-jet-azure we have containerservice/v1alpha2 API group. Now we have two resources in this API group and also we have two `terraformed.go` files: 

- `apis/containerservice/v1alpha2/zz_kubernetescluster_terraformed.go`
- `apis/containerservice/v1alpha2/zz_kubernetesclusternodepool_terraformed.go`

With this PR, we will generate a single `zz_generated.terraformed.go` for an API group.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

- Run make generate for provider-jet-azure and ensured that a single zz_generated.terraformed.go file is generated for the whole group.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: pkg/pipeline/terraformed.go [hunk 1]
```diff
@@ -49,30 +48,35 @@ type TerraformedGenerator struct {
 }
 
 // Generate writes generated Terraformed interface functions
-func (tg *TerraformedGenerator) Generate(cfg *config.Resource, parametersTypeName string) error {
+func (tg *TerraformedGenerator) Generate(cfgs map[*config.Resource]string, apiVersion string) error {
```
Comment (muvaf) [inline pkg/pipeline/terraformed.go:51]: nitpick: if the only reason to use a map is to couple type name with the configuration, you can consider coming up with a struct type to transit the information and then send an array of that object. Something like the following:
```go
type TerraformedInput struct { // maybe private?
  config.Resource
  ParametersTypeName string
}
```

**Review by muvaf — APPROVED** [review]
LGTM! Left some comments, should be good to go after addressing them. Thanks @sergenyalcin , I hadn't thought it'd be that easy to get this implemented!

File: pkg/pipeline/terraformed.go [hunk 2]
```diff
@@ -49,30 +48,35 @@ type TerraformedGenerator struct {
...
-			"Fields": cfg.Sensitive.GetFieldPaths(),
-		},
-		"LateInitializer": map[string]interface{}{
-			"IgnoredFields": cfg.LateInitializer.GetIgnoredCanonicalFields(),
-		},
+		"APIVersion": apiVersion,
 	}
-
-	filePath := filepath.Join(tg.LocalDirectoryPath, fmt.Sprintf("zz_%s_terraformed.go", strings.ToLower(cfg.Kind)))
+	var resources []map[string]interface{} //nolint:prealloc
```
[the remark this answers is not in the document]
Comment (sergenyalcin) [reply pkg/pipeline/terraformed.go]: Done!

File: pkg/pipeline/terraformed.go [hunk 3]
```diff
@@ -49,30 +48,35 @@ type TerraformedGenerator struct {
...
+			},
+			"Sensitive": map[string]interface{}{
+				"Fields": cfg.Sensitive.GetFieldPaths(),
+			},
+			"LateInitializer": map[string]interface{}{
+				"IgnoredFields": cfg.LateInitializer.GetIgnoredCanonicalFields(),
+			},
+		})
+	}
+	vars["Resources"] = resources
```
[the remark this answers is not in the document]
Comment (sergenyalcin) [reply pkg/pipeline/terraformed.go:84]: Done!

File: pkg/pipeline/terraformed.go (hunk 1 above)
Comment (sergenyalcin) [reply pkg/pipeline/terraformed.go]: Done!

**Pull request merged**